src/seedu/addressbook/AddressBook.java

```diff
@@ -306,13 +311,49 @@ private static void setupDefaultFileForStorage() {
 
     /**
      * Returns true if the given file is acceptable.
-     * The file path is acceptable if it ends in '.txt'
+     * The file path is acceptable if it does not contain any reserved string,
+     * contains a valid file name and valid path. 
      * TODO: Implement a more rigorous validity checking.
      */
-    private static boolean isValidFilePath(String filePath) {
-        return filePath.endsWith(".txt");
+    protected static boolean isValidFilePath(String filePath) {
+        return isValidPath(filePath) && isValidFileName(filePath);
     }
```

Comment:
This method should be in a utility class instead? It is not specific to AddressBook class.